bluefog/torch/optimizers.py

```diff
@@ -711,16 +757,21 @@ def DistributedPullGetOptimizer(optimizer, model):
         (optimizer.__class__,),
         dict(_DistributedBluefogOptimizer.__dict__),
     )
-    return cls(optimizer.param_groups, model, pull_style=True)
+    return cls(optimizer.param_groups, model, num_steps_per_communication, pull_style=True)
 
 
-def DistributedBluefogOptimizer(optimizer, model):
+def DistributedBluefogOptimizer(optimizer, model,
+                                num_steps_per_communication=1):
     """An distributed optimizer that wraps another torch.optim.Optimizer with
     pull model average through bf.win_put ops.
 
     Arguments:
         optimizer: Optimizer to use for computing gradients and applying updates.
         model: The model or a list of models you want to train with.
+        num_steps_per_communication: Number of expected step() function calls before each
+                                     communication. This allows local model parameter updates
+                                     per num_steps_per_communication before reducing them over
+                                     distributed computation resources.
 
```

Comment:
Add examples of how to correctly use num_steps_per_communication in two scenarios.

Reply:
Done.